### PR TITLE
Add Ord implementation for BoundedVec

### PR DESCRIPTION
**Pull request by kianenigma**
interestingly other bounded types already had it, this one was forgotten.

### CODE REVIEW

**Comment by gilescope** [comment]
can we derive PartialOrd here?

File: frame/support/src/storage/bounded_vec.rs [hunk 1]
```diff
@@ -57,6 +57,18 @@ impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<WeakBoundedVec<T, S>>
 }
 impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedSlice<'a, T, S> {}
 
+impl<T: PartialOrd, Bound: Get<u32>> PartialOrd for BoundedVec<T, Bound> {
```
Comment (bkchr) [inline frame/support/src/storage/bounded_vec.rs:60]:
```suggestion
impl<T: PartialOrd, Bound> PartialOrd for BoundedVec<T, Bound> {
```

File: frame/support/src/storage/bounded_vec.rs [hunk 2]
```diff
@@ -57,6 +57,18 @@ impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<WeakBoundedVec<T, S>>
...
+	fn partial_cmp(&self, other: &Self) -> Option<sp_std::cmp::Ordering> {
+		self.0.partial_cmp(&other.0)
+	}
+}
+
+impl<T: Ord, Bound: Get<u32>> Ord for BoundedVec<T, Bound> {
```
Comment (bkchr) [inline frame/support/src/storage/bounded_vec.rs:66]:
```suggestion
impl<T: Ord, Bound> Ord for BoundedVec<T, Bound> {
```

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge

File: frame/support/src/storage/bounded_vec.rs [hunk 3]
```diff
@@ -57,6 +57,18 @@ impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<WeakBoundedVec<T, S>>
 }
 impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<Vec<T>> for BoundedSlice<'a, T, S> {}
 
+impl<T: PartialOrd, Bound> PartialOrd for BoundedVec<T, Bound> {
```
Comment (bkchr) [inline frame/support/src/storage/bounded_vec.rs:60]:
```suggestion
impl<T: PartialOrd, Bound: Get<u32>> PartialOrd for BoundedVec<T, Bound> {
```

File: frame/support/src/storage/bounded_vec.rs [hunk 4]
```diff
@@ -57,6 +57,18 @@ impl<'a, T: Encode + Decode, S: Get<u32>> EncodeLike<WeakBoundedVec<T, S>>
...
+	fn partial_cmp(&self, other: &Self) -> Option<sp_std::cmp::Ordering> {
+		self.0.partial_cmp(&other.0)
+	}
+}
+
+impl<T: Ord, Bound> Ord for BoundedVec<T, Bound> {
```
Comment (bkchr) [inline frame/support/src/storage/bounded_vec.rs:66]:
```suggestion
impl<T: Ord, Bound: Get<u32>> Ord for BoundedVec<T, Bound> {
```

**Comment by bkchr** [comment]
bot merge

**Pull request merged**